src/ui-test/tests/jbang_user_preference.test.ts

```diff
@@ -0,0 +1,169 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License", destination); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { expect } from 'chai';
+import path = require('path');
+import { WebDriver, VSBrowser, Marketplace, By, EditorView, Workbench, InputBox, BottomBarPanel, TerminalView } from 'vscode-uitests-tooling';
+import * as pjson from '../../../package.json';
+import * as fs from 'fs-extra';
+
+describe('JBang user preference version set test', function () {
+    this.timeout(60000);
+
+    let driver: WebDriver;
+    let input: InputBox;
+    let terminalView: TerminalView;
+    let DEFAULT_JBANG: string;
+
+    const RESOURCES: string = path.resolve('src', 'ui-test', 'resources');
+
+    const FILENAME = 'test.camel.xml';
+    const OLDER_JBANG_VERSION = '3.20.5';
+
+    before(async function () {
+        this.timeout(200000);
+
+        driver = VSBrowser.instance.driver;
+        await VSBrowser.instance.openResources(RESOURCES);
+        await VSBrowser.instance.waitForWorkbench();
+
+        const marketplace = await Marketplace.open();
+        await driver.wait(async function () {
+            return await extensionIsActivated(marketplace);
+        }, 150000, `The LSP extension was not activated after ${this.timeout} sec.`);
+
+        DEFAULT_JBANG = await getJBangVersion();
+    });
+
+    after(async function () {
+        await setJBangVersion(DEFAULT_JBANG);
+    });
+
+    describe('Different JBang versions', function () {
+        afterEach(async function () {
+            await new EditorView().closeAllEditors();
+            deleteFile(FILENAME);
+        });
+
+        it('Default version', async function () {
+            await setJBangVersion(DEFAULT_JBANG);
+
+            await new Workbench().executeCommand('Camel: Create a Camel Route using XML DSL');
```

Comment:
init and wait for file is created could be moved to method and reused in both tests